File: Aula2/Java/Mago.java
public class Mago extends Personagem {
    private int mana;

    public Mago(String nome, int vida, int forca, int defesa) {
        super(nome, vida, forca, defesa);
        this.mana = 50;
    }

    public int getMana() {
        return mana;
    }

    public void setMana(int mana) {
        if (mana < 0) {
            throw new IllegalArgumentException("A mana não pode ser negativa.");
        }

        this.mana = mana;
    }

    @Override
    public void ficha() {
        super.ficha();
        System.out.println("Mana: " + mana);
    }
    @Override
    public void habilidadeEspecial(Personagem alvo) {
        int custoMana = 10;
    
        if (this.mana < custoMana) {
            System.out.println(this.nome + " não tem mana suficiente para conjurar!");
            return;
        }
    
        this.mana -= custoMana;
        int danoMagico = this.forca * 2;
    
        System.out.println(this.nome + " usa rajada arcana!");
        alvo.receberDano(danoMagico);
    }
}

File: Aula2/Python/Mago.py
from PersonagemPy import Personagem


class Mago(Personagem):
    def __init__(self, nome, vida, forca, defesa=0):
        super().__init__(nome, vida, forca, defesa)
        self.__mana = 50

    @property
    def mana(self):
        return self.__mana

    @mana.setter
    def mana(self, valor):
        if valor < 0:
            raise ValueError("A mana não pode ser negativa.")

        self.__mana = valor

    def ficha(self):
        super().ficha()
        print(f"Mana: {self.__mana}")